cirq-core/cirq/contrib/acquaintance/executor.py

```diff
@@ -100,6 +115,65 @@ def optimization_at(
         )
 
 
+class StrategyExecutorTransformer:
+    """Executes an acquaintance strategy."""
+
+    def __init__(self, execution_strategy: ExecutionStrategy) -> None:
+        self.execution_strategy = execution_strategy
+        self.mapping = execution_strategy.initial_mapping.copy()
+
+    def __call__(
+        self, circuit: 'cirq.Circuit', context: Optional['cirq.TransformerContext'] = None
+    ) -> circuits.AbstractCircuit:
+        """Executes an acquaintance strategy using primitive map function and
+          mutates initial mapping.
+
+        Args:
+            circuit: Input circuit to transform.
+            context: `cirq.TransformerContext` storing common configurable
+              options for transformers.
+
+        Raises:
+            ValueError: if execution_strategy is None
+            TypeError: if execution_strategy neither an instance of
+              AcquaintanceOpportunityGate or PermutationGate
```

Comment:
These value errors are checking the type of `execution_strategy`; so they should be raised in the `__init__` method itself when instantiating the class with incorrect parameters.